components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt

```diff
@@ -270,12 +271,19 @@ class AppLinksUseCases(
 
         internal val ALWAYS_DENY_SCHEMES: Set<String> = setOf("jar", "file", "javascript", "data", "about")
 
+        @Suppress("TooGenericExceptionCaught")
         @VisibleForTesting
         internal fun safeParseUri(uri: String, flags: Int): Intent? {
             return try {
                 Intent.parseUri(uri, flags)
-            } catch (e: URISyntaxException) {
-                null
+            } catch (e: Exception) {
+                when (e) {
+                    is URISyntaxException, is NumberFormatException -> {
+                        Logger.error("failed to parse URI", e)
+                        null
+                    }
+                    else -> throw e
+                }
```

Comment:
Is this really simpler than just two `catch()` blocks? In addition with the suppress annotations this seems so verbose now. :)

Reply:
True, my original change was with 2 `catch()`.  End up choosing this because there's another one above that does it this way.  But now I think about it, we're not performing any `failedAction`.  I'll switch back to 2 `catch()`.  Thanks